File: Assignment/app/src/main/java/com/assign/assignment/MainActivity.java
package com.assign.assignment;

import android.app.Activity;
import android.app.FragmentTransaction;
import android.os.Bundle;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        ListFragment fragment = (ListFragment) getFragmentManager().findFragmentById(R.id.fragment_container);

        if (fragment == null) {
            FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
            fragmentTransaction.add(R.id.fragment_container, new ListFragment());
            fragmentTransaction.commit();
        }

    }
}

File: Assignment/app/src/main/java/com/assign/assignment/MainActivity.kt
package com.assign.assignment

import android.app.Activity
import android.os.Bundle

class MainActivity : Activity() {
   private var fragment: ListFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fragment = fragmentManager.findFragmentById(R.id.fragment_container) as ListFragment
        if (fragment == null) {
            val fragmentTransaction = fragmentManager.beginTransaction()
            fragmentTransaction.add(R.id.fragment_container, ListFragment()).commit()
        }
    }
}
